### PR TITLE
Update doc for CA1501 to mention configurability of the inheritance check

### LINKED ISSUE #4984: CA1501 - Add configurability section for user specific exclusion
Starting with release 3.x of fxcop, a new user option has been added to allow exclusion of specific types/namespaces from the hierarchy count. The default value of this option has been set to `System.*` to allow exclusion of all .net types. This is especially nice for types deriving from `UserControl` (WPF or WinForms) as they were all faulty by default. Note that this default value cannot be overriden.

(see https://github.com/dotnet/roslyn-analyzers/pull/3233/files#diff-1b5cdc4937a8cdfc79545945ad6c0783R478-R503)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 971b064d-7b0a-df32-00fc-99851026122a
* Version Independent ID: 7298bd5e-ba4b-ae59-a53d-293f9790a73b
* Content: [CA1501: Avoid excessive inheritance - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1501?view=vs-2019)
* Content Source: [docs/code-quality/ca1501.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1501.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4984

This new feature is not yet publicly released. Please sync-up with @mavasani for when to merge this PR.

### CODE REVIEW

**Comment by PRMerger6** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1501.md [hunk 1]
```diff
@@ -49,3 +49,20 @@ The following example shows a type that violates the rule:
 
 [!code-csharp[FxCop.Maintainability.ExcessiveInheritance#1](../code-quality/codesnippet/CSharp/ca1501-avoid-excessive-inheritance_1.cs)]
 [!code-vb[FxCop.Maintainability.ExcessiveInheritance#1](../code-quality/codesnippet/VisualBasic/ca1501-avoid-excessive-inheritance_1.vb)]
+
+## Configurability
```
Comment (mavasani) [inline docs/code-quality/ca1501.md:53]: Please add a small sentence about configurability at the end of the `Cause` section above in the document, with a hyperlink to this section. Similar to the last sentence in https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1802?view=vs-2019#cause

Comment (mavasani) [reply docs/code-quality/ca1501.md:53]: Also all the existing pages have "Configurability" section above the "Example" section. Please retain this order.

**Comment by ktoliver** [comment]
Hi @Evangelink - Could you make the revisions that @mavasani requests? Thanks.

**Comment by PRMerger20** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Evangelink** [comment]
Hi @ktoliver, Let me know if there is anything else to do. Also, feel free to push a change if there is anything to change.

**Comment by Mikejo5000** [comment]
#sign-off

**Pull request merged**